src/pkg/caendr/caendr/models/datastore/entity.py

```diff
@@ -388,6 +388,45 @@ def query_ds_unique(cls, key, val, required=False, safe=False):
       raise NonUniqueEntity( cls.kind, key, val, matches )
 
 
+  @classmethod
+  def query_ds_not_deleted(cls, key, val, required=False):
+
+    # Run query with given key and val
+
+    matches = cls.query_ds(filters=[(key, '=', val)])
+
+    # TODO: Filter out any elements in "matches" that are deleted
```

Comment:
Very nitpicky point, but might want to clean up the spacing here & remove this `TODO` message.  I know we're pretty lax about `TODO`s, but since you do actually implement what it describes in this PR, I'd recommend cleaning it up.